autotest/ogr/ogr_gpkg.py

```diff
@@ -1601,6 +1601,65 @@ def test_ogr_gpkg_srs_non_consistent_with_official_definition():
     gdal.Unlink('/vsimem/ogr_gpkg_20.gpkg')
 
 
+def test_ogr_gpkg_write_srs_undefined_geographic():
+
+    gdal.Unlink('/vsimem/ogr_gpkg_srs_undefined_geographic.gpkg')
+
+    ds = gdaltest.gpkg_dr.CreateDataSource('/vsimem/ogr_gpkg_srs_undefined_geographic.gpkg')
+    assert ds is not None
+
+    # Check initial default SRS entries in gpkg_spatial_ref_sys
+    sql_lyr = ds.ExecuteSQL("SELECT COUNT(*) FROM gpkg_spatial_ref_sys")
+    gpkg_spatial_ref_sys_total = sql_lyr.GetNextFeature().GetField(0)
+    assert gpkg_spatial_ref_sys_total == 3 # SRS with IDs: -1, 0, 4326
+    ds.ReleaseResultSet(sql_lyr)
+
+    srs= osr.SpatialReference()
+    srs.SetFromUserInput('GEOGCS["Undefined geographic SRS",DATUM["unknown",SPHEROID["unknown",6378137,298.257223563]],PRIMEM["Greenwich",0],UNIT["degree",0.0174532925199433,AUTHORITY["EPSG","9122"]],AXIS["Latitude",NORTH],AXIS["Longitude",EAST]]')
+    lyr = ds.CreateLayer('srs_test_layer', geom_type=ogr.wkbPoint, srs=srs)
+    assert lyr.GetSpatialRef().ExportToWkt().find('Undefined geographic SRS') >= 0
+
+    # Check no new SRS entries have been inserted into gpkg_spatial_ref_sys
+    sql_lyr = ds.ExecuteSQL("SELECT COUNT(*) FROM gpkg_spatial_ref_sys")
```

Comment:
this test would be a bit more convincing after closing and reopening the dataset (in case we would do defered writing). Same for other test

Reply:
Good point, thanks. Done.